File: OrbitalMechanics/Helpers/OrbitHelper.vb
Imports Core.Classes

Public Class OrbitHelper

    Public Function CalculateTotalMass(massOfPrimary As Mass, massOfSatellite As Mass) As Mass
        Return Mass.FromKilograms(massOfPrimary.Kilograms + massOfSatellite.Kilograms)
    End Function

    Public Function CalculatePeriod(totalMass As Mass, semiMajorAxis As Distance) As TimeSpan
        Dim daysSquared = ((4 * Math.PI * Math.PI * Math.Pow(semiMajorAxis.AstronomicalUnits, 3)) / (Math.Pow(Constants.GaussianGravitationalConstant, 2) * totalMass.SolarMasses))

        Return TimeSpan.FromDays(Math.Pow(daysSquared, 0.5))
    End Function

End Class

File: space-game/OrbitalMechanics/Helpers/OrbitHelper.cs
using Core.Classes;
using System;

namespace OrbitalMechanics.Helpers
{
    public class OrbitHelper
    {
        public Mass CalculateTotalMass(Mass massOfPrimary, Mass massOfSatellite)
        {
            return Mass.FromKilograms(massOfPrimary.Kilograms + massOfSatellite.Kilograms);
        }

        public TimeSpan CalculatePeriod(Mass totalMass, Distance semiMajorAxis)
        {
            double daysSquared = ((4 * Math.PI * Math.PI * Math.Pow(semiMajorAxis.AstronomicalUnits, 3)) / (Math.Pow(Constants.GaussianGravitationalConstant, 2) * totalMass.SolarMasses));

            return TimeSpan.FromDays(Math.Pow(daysSquared, 0.5));
        }
    }
}
